### PR TITLE
Update library

### PR DESCRIPTION
**Pull request by antal-bukos**
- decaffeinate
- add eslint, flow and prettier
- remove dependency on underscore js
- update used library versions
- change library behavior to not convert functional components to class components, thus allowing the usage of react hooks

### CODE REVIEW

File: jest-preprocessor.js [hunk 1]
```diff
@@ -1,12 +1,18 @@
...
+  process: function(src, path, ...rest) {
     if (path.match(/\.(s?css)/)) {
       return;
     }
-    if (coffee.helpers.isCoffee(path)) {
-      return coffee.compile(src, {'bare': true});
+    if (
+      path.indexOf('node_modules') === -1 ||
+      path.indexOf('bluecore-ui-kit') > -1 ||
+      path.indexOf('bluecore-api-lib') > -1
```
Comment (istvan-hevele) [inline jest-preprocessor.js]: I think `bluecore-ui-kit` and `bluecore-api-lib` is unnecessary here.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "bluecore-classnames",
-  "version": "0.4.0",
+  "version": "1.0.0",
```
Comment (istvan-hevele) [inline package.json]: 👍

File: jest-preprocessor.js (hunk 1 above)
Comment (antal-bukos) [reply jest-preprocessor.js]: I think you're right 😄

File: .npmignore [hunk 3]
```diff
@@ -1,5 +1,11 @@
 __tests__
 examples
-coffeelint.json
 jest-preprocessor.js
 webpack.*
+.idea
```
[the remark this answers is not in the document]
Comment (antal-bukos) [reply .npmignore]: yes

**Pull request merged**